This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:

Copyright (c) 2025 Qualcomm Technologies, Inc. All rights reserved.
SPDX-License-Identifier: BSD-3-Clause-Clear

From 96129709ee2498f0fb2a9ddf03836d88f90e0ce1 Mon Sep 17 00:00:00 2001
From: Jing Wang <jingwa@qti.qualcomm.com>
Date: Thu, 22 May 2025 17:43:29 +0800
Subject: [PATCH] utils: tinyplay: check fopen return value to prevent
 segmention fault

Upstream-Status: Pending
Signed-off-by: Jing Wang <jingwa@qti.qualcomm.com>
---
 utils/tinyplay.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/utils/tinyplay.c b/utils/tinyplay.c
index d617074..42df1aa 100644
--- a/utils/tinyplay.c
+++ b/utils/tinyplay.c
@@ -33,6 +33,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <errno.h>
 
 #define OPTPARSE_IMPLEMENTATION
 #include "optparse.h"
@@ -184,16 +185,15 @@ static int ctx_init(struct ctx* ctx, struct cmd *cmd)
         ctx->file = stdin;
     } else {
         ctx->file = fopen(cmd->filename, "rb");
+        if (ctx->file == NULL) {
+            fprintf(stderr, "failed to open '%s', %s\n", cmd->filename, strerror(errno));
+            return -1;
+        }
         fseek(ctx->file, 0L, SEEK_END);
         ctx->file_size = ftell(ctx->file);
         fseek(ctx->file, 0L, SEEK_SET);
     }
 
-    if (ctx->file == NULL) {
-        fprintf(stderr, "failed to open '%s'\n", cmd->filename);
-        return -1;
-    }
-
     if (is_wave_file(cmd->filetype)) {
         if (parse_wave_file(ctx, cmd->filename) != 0) {
             fclose(ctx->file);
-- 
2.34.1

